holoviews/operation/datashader.py

```diff
@@ -1725,3 +1725,125 @@ class directly_connect_edges(_connect_edges, connect_edges):
 
     def _bundle(self, position_df, edges_df):
         return connect_edges.__call__(self, position_df, edges_df)
+
+
+def identity(x): return x
+
+class inspect_points(Operation):
+    """
+    Given datashaded aggregate (Image) output, return a single
+    (hoverable) point at the sample closest to the cursor.
+    """
+
+    pixels = param.Integer(default=3, doc="""
+       Number of pixels in data space around the cursor point to search
+       for hits in. The hit within this box mask that is closest to the
+       cursors position is displayed.""")
+
+    null_value = param.Number(default=0, doc="""
+       Value of raster which indicates no hits. For instance zero for
+       count aggregator (default) and commonly NaN for other (float)
+       aggregators. For RGBA images, the alpha channel is used which means
+       zero alpha acts as the null value.""")
+
+    value_bounds = param.NumericTuple(default=None, length=2, allow_None=True, doc="""
+       If not None, a numeric bounds for the pixel under the cursor in
+       order for hits to be computed. Useful for count aggregators where
+       a value of (1,1000) would make sure no more than a thousand
+       samples will be searched.""")
+
+    hits = param.DataFrame(default=pd.DataFrame(), allow_None=True)
+
+    point_count = param.Integer(default=1, doc="""
+       Maximum number of points to display within the mask of size
+       pixels. Points are prioritized by distance from the cursor
+       point. This means that the default value of one shows the single
+       closest sample to the cursor.""")
+
+    hits_transformer = param.Callable(default=identity, doc="""
+       Function that can transform the hits dataframe before it is set
+       on the hits parameter. When no hits occur, the value passed to
+       the callable is None. This can be used to do multiple common
+       tasks such as 1) subselecting available columns 2) limiting the
+       maximum number of hit rows returned 3) performing joins with
+       other data sources 4) returning a dataframe with appropriate
+       columns and dtypes when no hits occur. The transformed dataframe
+       is also what is passed to the Points object.""")
+
+    points_transformer = param.Callable(default=identity, doc="""
+      Function that transforms the hits dataframe""")
+
+    vdims = param.List(default=[], doc="""
+       The vdims of the returned Point element drawn from the source
+       dataframe. If a points_transformer is used, these columns need to
+       in the return value.""")
+
+    # Stream values and overrides
+    streams = param.List(default=[PointerXY])
+    x = param.Number(default=0)
+    y = param.Number(default=0)
+
+    def _process(self, raster, key=None):
+        no_match = Points(self.p.hits_transformer(None), vdims=self.p.vdims)
+        try:
+            if isinstance(raster, RGB):
```

Comment:
Shouldn't happen for datashaded images but not all RGBs have an alpha channel.

Reply:
Good point. I'll make sure to check the channel number. If alpha is missing, that would be an error imho as I don't know what else you could do (maybe say white or black are null values but then you wouldn't have a single scalar, unless you use a hex color or something).

Reply:
I guess that could be a warning, and then just treat it as if the count is non-zero and non-NaN.

Reply:
In other words, disable the sparseness optimization and always run the spatial queries?

Reply:
Right.